### ISSUE TITLE
Support for TCP as transport protocol

### ISSUE THREAD

**Issue by powershellshock** [poshsecurity/Posh-SYSLOG#6]
It would be great to add support for TCP in addition to UDP. Though TCP transport for syslog is not formally defined as a standard, it is widely supported by numerous solutions. RFC 6587, while not a standard describes how TCP has been used as a transport for syslog messages, which would enable us to develop this support.

**Comment by powershellshock** [comment]
I have already done most of the work to implement this by modifying the existing Send-SyslogMessage cmdlet. I have added some parameters to enable this and also modified existing ones, but created aliases to ensure backward compatibility. Please add me as a contributor, so that I may submit these changes for review. We may want to add a branch for this until the testing is more thorough. (As I said, the code is pretty much done already, but I think more testing and review is appropriate.)

**Comment by kjacobsen** [comment]
Hi Jared,

TCP support would be great, it has been something I have been thinking about for a while (but haven't had the need).

If you create a pull request, I can then work with you to review the changes and include them into the module.

**Comment by powershellshock** [comment]
Ok. So far, I have only contributed to a Github project that I own, so I don’t have much experience contributing to a project administered by someone else. Do you just want me to create a pull request for the master branch?

Regards,
Jared Poeppelman, CISSP | Cybersecurity, Cloud Security, Identity, Powershell | [Description: MSFT_logo_Gray DE sized SIG1.png] <http://www.microsoft.com/>
• jpoeppel@microsoft.com<mailto:jpoeppel@microsoft.com>   • 513-295-4585
Ask me about Powershell for Cloud App Security<https://github.com/Microsoft/Cloud-App-Security>

**Comment by powershellshock** [comment]
I cannot create a pull request in any branches because I am not an admin on the project. Please create a dev branch and grant me access to make commits to that. Then, I can submit a pull request for you to pull into master from there, when we are ready. I also have a CEF (Common Event Format) message cmdlet (Send-CEFSyslogMessage) almost done, too. From that, I will be able to quickly add a LEEF (Log Event Extended Format) message cmdlet (Send-LEEFSyslogMessage), as well. Those will be helpful for ArcSight and QRadar (and other SIEM) users.

**Comment by powershellshock** [comment]
Deleted my previous comments. (I am still learning GitHub and have now resolved my forking issues ;-) .) I will submit a pull request soon.

**Comment by kjacobsen** [comment]
Hi, 

Reviewing the PR now.

I have created a new branch, version3, and will merge the PR into that first. From there I will work through any changes or tweaks that need to be made. Once that is complete, we will push into Master and then onto PowerShell Gallery.

I  am going to increment the version number to 3.0.0 for these changes.

**Comment by kjacobsen** [comment]
I have just merged the version3 branch into master as part of #8
